prow/plugins/heart/heart.go

```diff
@@ -73,3 +83,34 @@ func handle(gc githubClient, log *logrus.Entry, ic github.IssueCommentEvent) err
 		ic.Comment.ID,
 		reactions[rand.Intn(len(reactions))])
 }
+
+func handlePR(gc githubClient, log *logrus.Entry, pre github.PullRequestEvent) error {
+	// Only consider newly opened PRs
+	if pre.Action != "opened" {
+		return nil
+	}
+
+	changes, err := gc.GetPullRequestChanges(pre.PullRequest)
+	if err != nil {
+		return err
+	}
+
+	// Smile at any change that adds to OWNERS files
+	ownersFileAdditions := false
+	for _, change := range changes {
+		_, filename := filepath.Split(change.Filename)
+		if filename == ownersFilename && change.Additions > 0 {
+			ownersFileAdditions = true
```

Comment:
Maybe it's just me, but I find that control-flow pretty confusing and try to avoid it in general.

Could you: 
- Remove the extra local variable `ownersFileAdditions`,
- Remove the `if ownersFileAdditions` below,
- Move the return directly in the loop?